File: WeatherWidget/src/main/java/org/voegtle/weatherwidget/util/NotificationTask.java
package org.voegtle.weatherwidget.util;

import android.content.Context;
import android.os.AsyncTask;
import android.widget.Toast;

public class NotificationTask extends AsyncTask<Void, Void, Void> {
  private String message;
  private Context context;

  public NotificationTask(Context context, String message) {
    this.message = message;
    this.context = context;
  }

  @Override
  protected Void doInBackground(Void... voids) {
    return null;
  }

  @Override
  protected void onPostExecute(Void aVoid) {
    int duration = Toast.LENGTH_SHORT;
    Toast toast = Toast.makeText(context, message, duration);
    toast.show();

  }
}

File: src/main/java/org/voegtle/weatherwidget/util/NotificationTask.kt
package org.voegtle.weatherwidget.util

import android.content.Context
import android.os.AsyncTask
import android.widget.Toast

class NotificationTask(private val context: Context, private val message: String) : AsyncTask<Void, Void, Void>() {

  override fun doInBackground(vararg voids: Void): Void? {
    return null
  }

  override fun onPostExecute(aVoid: Void?) {
    val duration = Toast.LENGTH_SHORT
    val toast = Toast.makeText(context, message, duration)
    toast.show()
  }
}
